sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java

```diff
@@ -50,57 +50,110 @@
         this.service = service;
     }
 
-    Mono<Response<DocumentResultCollection<AnalyzeSentimentResult>>> analyzeSentimentBatchWithResponse(
+    /**
+     * Helper function that analyzes a batch of documents and returns {@link TextAnalyticsPagedFlux} that is a paged
+     * flux contains {@link AnalyzeSentimentResult}.
+     *
+     * @param textInputs A batch of input texts.
+     * @param options The request options, such as the training model version and to show statistics.
+     *
+     * @return {@link TextAnalyticsPagedFlux} of {@link AnalyzeSentimentResult}.
+     */
+    TextAnalyticsPagedFlux<AnalyzeSentimentResult> analyzeSentimentBatch(Iterable<TextDocumentInput> textInputs,
+        TextAnalyticsRequestOptions options) {
+        Objects.requireNonNull(textInputs, "'textInputs' cannot be null.");
+        try {
+            return new TextAnalyticsPagedFlux<>(() -> (continuationToken, pageSize) -> withContext(context ->
+                service.sentimentWithRestResponseAsync(
+                    new MultiLanguageBatchInput().setDocuments(Transforms.toMultiLanguageInput(textInputs)),
+                    options == null ? null : options.getModelVersion(),
+                    options == null ? null : options.isStatisticsShown(), context)
+                    .doOnSubscribe(ignoredValue ->
+                        logger.info("A batch of text sentiment input - {}", textInputs.toString()))
```

Comment:
Is the idea that when the user activates logging, it will see all of this?
Do we really need to print all the input?
Also, `text sentiment input` sounds weird. The text doesn't include the sentiment, the service is going to analyze the text to get a sentiment. So maybe, `Batch of documents to analyze`

Reply:
It is a code style for using the reactor. When subscribe happens, we want to know the inputs.
will change to `A batch of documents to analyze`